app/main/views/suppliers.py

```diff
@@ -418,7 +419,19 @@ def update_supplier_on_framework(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
-    interest_record.on_framework = update_json['onFramework']
+    if "onFramework" in update_json:
+        interest_record.on_framework = update_json["onFramework"]
+    if "prefillDeclarationFromFrameworkSlug" in update_json:
+        if update_json["prefillDeclarationFromFrameworkSlug"] is not None:
+            prefill_declaration_from_framework_id = db.session.query(SupplierFramework.framework_id).filter(
+                SupplierFramework.framework.has(Framework.slug == update_json["prefillDeclarationFromFrameworkSlug"]),
+                SupplierFramework.supplier_id == supplier.supplier_id,
+            ).scalar()
+            if prefill_declaration_from_framework_id is None:
+                abort(400, "Unknown framework slug '{}'".format(update_json["prefillDeclarationFromFrameworkSlug"]))
```

Comment:
Oh that's interesting - I wonder how my change to this message got squeezed out in my rebase. It was supposed to be `Supplier hasn't registered interest in a framework with slug {}`, which is a bit of a cheat because what I'm doing is using the same error for both "unknown framework" and "not registered interest".